pkg/cover/backend/elf.go

```diff
@@ -196,3 +176,43 @@ func elfReadModuleCoverPoints(target *targets.Target, module *Module, info *symb
 	}
 	return pcs, nil
 }
+
+func alignUp(addr, align uint64) uint64 {
+	if align == 0 {
+		return addr
+	}
+	return (addr + align - 1) & ^(align - 1)
+}
+
+// Calculate the offset of the module .text section in the kernel memory.
+// /proc/modules only contains the base address, which corresponds to the
+// beginning of the first code section, but that section does not have to
+// be .text (e.g. on Android it may be .plt).
+// The offset is calculated by summing up the aligned sizes of sections
+// that:
+//   - precede .text;
+//   - are not .init sections;
+//   - have the same section flags as .text.
+func elfGetModuleOffset(path string) uint64 {
+	file, err := elf.Open(path)
+	if err != nil {
+		return 0
+	}
+	defer file.Close()
+	ts := file.Section(".text")
+	if ts == nil {
+		return 0
+	}
+	off := uint64(0)
+	textFlags := ts.SectionHeader.Flags
+	for _, s := range file.Sections {
+		if s.Flags == textFlags && !strings.HasPrefix(s.SectionHeader.Name, ".init") {
```

Comment:
We also need to ignore ".exit" sections:
https://elixir.bootlin.com/linux/latest/source/kernel/module/main.c#L1498

It's strange that kernel ignores them only if CONFIG_MODULE_UNLOAD is enabled. Then if a module has it, but kernel does not have CONFIG_MODULE_UNLOAD, then it will load it into text segment for no reason. But probably modules config should match kernel config and module linker script removes .exit sections when CONFIG_MODULE_UNLOAD is not enabled.